### ISSUE TITLE
Editing multiple times discards previous edits

### ISSUE THREAD

**Issue by tdilo** [openchargemap/ocm-system#143]
When editing the same charging station entry twice in short succession (i.e. before the change is approved by a country editor) the second edit seems to discard all changes made in the first one, unless all the details are explicitly entered again. I think this also happens when simply adjusting the location, in which case it is not obvious that the other unapproved edits get lost.

The edit page should either load the previous edits made, or calculate the differences to the original unmodified entry and create a second changeset to be approved by a country editor.

**Comment by webprofusion-chrisc** [comment]
Yes, this is correct. Each time you make an edit you are working with the current published version. What happens is a snapshot of your changes is saved for review and a diff is presented to the editor for review. If they approve these then your version is applied. 

So the options are:
1) load the most recent edit by the same user (unless they are anonymous) in the editor so that the edits are cumulative, delete any previous edits by that user on Submit.
or
2) I've been contemplating apply user edits immediately but still let an editor review and revert them later. If nobody reviews an edit then the edit gets auto approved by system after a certain cut off. This saves edits from languishing but increases the risk that bad edits will be accepted. Note that we have a shortage of editors overall and rely on a few good people to do all the work (generally, that's not me).

**Comment by tdilo** [comment]
So far I've only had good experiences with the responsiveness of the editors. :+1:

I think in general it is more rewarding for new contributors when they can effect change without delay. Even if you got some feedback when a change is accepted, the effort spent to do the change is long past and the reward does not match up with it anymore. Immediately applying user edits therefore might improve user retention, especially in times when editors are busy otherwise and pending changes build up. However, I think this should only be applied to registered users. My feeling is that for anonymous edits, the review process should still stay in place as it is. Either way, being able to see the editing history of a charge point would probably be beneficial to make it easier to review changes.

Option 1 sounds good when the review process is kept as is. I think this is usually what you'd want when editing twice, for example when correcting a typo. At the very least, editing while pending edits are in the approval queue should display a big warning message.

**Comment by webprofusion-chrisc** [comment]
Thanks yes, loading current edits from the same user sounds sensible. 

The other interesting one is when two people edit the same site, currently if two edits happened from different users, the last accepted edit would completely overwrite the other one. It's not something the happens often.
